src/components/SelectInput/SelectInput.tsx

```diff
@@ -12,119 +12,104 @@ import './SelectInput.scss'; // Not a module because it requires :global styles
+    value: ValueType<OptionTypeBase>;
+  }
+}
+
+interface Props {
   /**
    * The id attribute of the input
    */
-  id: PropTypes.string.isRequired,
+  id: string;
   /**
-   * Visually hide the label
+   * Custom content to be displayed above the input. If the label is hidden, will be used to set aria-label attribute.
    */
-  hideLabel: PropTypes.bool,
+  label: string;
   /**
-   * Custom content to be displayed above the input. If the label is hidden, will be used to set aria-label attribute.
+   * Callback function to call on change event.
    */
-  label: PropTypes.string.isRequired,
+  onChange: (event: SimulatedEventPayload) => void;
+  /**
+   * Options for dropdown list
+   */
+  options: {
+    value: string;
+    label: string;
+  }[];
+  /**
+   * Visually hide the label
+   */
+  hideLabel?: boolean;
   /**
    * Additional classes to add
    */
-  className: PropTypes.string,
+  className?: string;
   /**
    * Placeholder
    */
-  placeholder: PropTypes.string,
+  placeholder?: string;
   /**
    * Mark the input field as invalid and display a validation message.
    * Pass a string or node to render a validation message below the input
    */
-  error: PropTypes.oneOfType([
-    PropTypes.bool,
-    PropTypes.string,
-    PropTypes.node,
-  ]),
+  error?: React.ReactNode;
   /**
    * If the input should be disabled and not focusable
    */
-  isDisabled: PropTypes.bool,
+  isDisabled?: boolean;
   /**
    * Determines if input is required or not. (Label will have an asterisk if required)
    */
-  isRequired: PropTypes.bool,
+  isRequired?: boolean;
   /**
    * Select 'name' attribute
    */
-  name: PropTypes.string,
-  /**
-   * Callback function to call on change event.
-   */
-  onChange: PropTypes.func.isRequired,
+  name?: string;
   /**
    * Callback function to call on focus event.
    */
-  onFocus: PropTypes.func,
+  onFocus?: (event: FocusEvent<HTMLElement>) => void;
   /**
    * Callback function to call on blur event.
    */
-  onBlur: PropTypes.func,
+  onBlur?: (event: FocusEvent<HTMLElement>) => void;
   /**
    * Autofocus select input on render
    */
-  autoFocus: PropTypes.bool,
+  autoFocus?: boolean;
   /**
    * Is multi select enabled
    */
-  isMulti: PropTypes.bool,
-  /**
-   * Options for dropdown list
-   */
-  options: PropTypes.arrayOf(PropTypes.shape({
-    value: PropTypes.string.isRequired,
-    label: PropTypes.string.isRequired,
-  })).isRequired,
+  isMulti?: boolean;
   /**
    * The value(s) of select
    */
-  value: PropTypes.oneOfType([
-    PropTypes.object,
-    PropTypes.arrayOf(PropTypes.object),
-  ]),
+  value?: any | any[];
 };
 
-const defaultProps = {
-  className: undefined,
-  placeholder: undefined,
-  error: false,
-  hideLabel: false,
-  isDisabled: false,
-  isRequired: false,
-  name: '',
-  onFocus: undefined,
-  onBlur: undefined,
-  autoFocus: false,
-  isMulti: false,
-  value: undefined,
-};
-
-const SelectInput = ({
+const SelectInput: React.FC<Props> = ({
```

Comment:
For Button, we import `FC` at the top, and then call it as a standalone. I don't have a strong opinion either way on how it's done, but it should be one or the other. Perhaps either @nathanyoung or @juanfabrega has an opinion on this?

```
// Button.tsx
import React, { FC } from 'react';

const Button: FC<Props> = ({
```

Reply:
I prefer a single React import statement personally, But no strong opinion either, let's just make all of them consistent. @ericyliu choose your favorite and enforce it on all our tsx files in this PR.

Reply:
yeah true, im ok with either way

Reply:
i'll change everything to a single React import